File: app/src/api/maintenance.ts
import { apiClient } from './client'

export interface StorageSummary {
  database_path: string
  database_exists: boolean
  database_size_mb: number
  artifact_dir: string
  artifact_dir_exists: boolean
  artifact_size_mb: number
  artifact_file_count: number
  backup_dir: string
  backup_count: number
  latest_backup: string | null
  peptide_count: number
  task_count: number
  reviewed_count: number
  shortlisted_count: number
  selected_for_synthesis_count: number
  disclaimer: string
}

export interface BackupItem {
  filename: string
  path: string
  size_mb: number
  created_at: string
}

export interface BackupsList {
  db_backups: BackupItem[]
  artifact_backups: BackupItem[]
  snapshots: BackupItem[]
}

export interface BackupResult {
  backup_path: string
  size_mb: number
  created_at: string
}

export interface SnapshotResult {
  snapshot_path: string
  size_mb: number
  created_at: string
  manifest: Record<string, any>
}

export interface RestoreResult {
  status: string
  message: string
  backup_path?: string
  pre_restore_backup?: string
}

export interface CleanupResult {
  dry_run: boolean
  files_to_delete: number
  total_size_mb: number
  older_than_days: number
  message: string
}

export interface ResetDemoResult {
  status: string
  message: string
  deleted_count: number
  pre_reset_backup?: string
}

export function getStorageSummary(): Promise<StorageSummary> {
  return apiClient.get('/v1/maintenance/storage-summary')
}

export function backupDatabase(): Promise<BackupResult> {
  return apiClient.post('/v1/maintenance/backup-database', {})
}

export function backupArtifacts(): Promise<BackupResult> {
  return apiClient.post('/v1/maintenance/backup-artifacts', {})
}

export function createProjectSnapshot(): Promise<SnapshotResult> {
  return apiClient.post('/v1/maintenance/create-project-snapshot', {})
}

export function listBackups(): Promise<BackupsList> {
  return apiClient.get('/v1/maintenance/backups')
}

export function restoreDatabase(payload: { backup_filename: string; confirm: boolean }): Promise<RestoreResult> {
  return apiClient.post('/v1/maintenance/restore-database', payload)
}

export function cleanupArtifacts(payload: { older_than_days: number; dry_run: boolean }): Promise<CleanupResult> {
  return apiClient.post('/v1/maintenance/cleanup-artifacts', payload)
}

export function resetDemoData(payload: { confirm: boolean; include_real_runs: boolean; include_review_data: boolean }): Promise<ResetDemoResult> {
  return apiClient.post('/v1/maintenance/reset-demo-data', payload)
}

File: backend/app/schemas/maintenance.py
from typing import Optional, List
from pydantic import BaseModel, Field
from datetime import datetime


class StorageSummaryOut(BaseModel):
    database_path: str
    database_exists: bool
    database_size_mb: float
    artifact_dir: str
    artifact_dir_exists: bool
    artifact_size_mb: float
    artifact_file_count: int
    backup_dir: str
    backup_count: int
    latest_backup: Optional[str] = None
    peptide_count: int
    task_count: int
    reviewed_count: int
    shortlisted_count: int
    selected_for_synthesis_count: int
    disclaimer: str


class BackupOut(BaseModel):
    backup_path: str
    size_mb: float
    created_at: str


class BackupListItem(BaseModel):
    filename: str
    path: str
    size_mb: float
    created_at: str


class BackupsOut(BaseModel):
    db_backups: List[BackupListItem]
    artifact_backups: List[BackupListItem]
    snapshots: List[BackupListItem]


class RestorePayload(BaseModel):
    backup_filename: str
    confirm: bool = False


class RestoreOut(BaseModel):
    status: str
    message: str
    backup_path: Optional[str] = None
    pre_restore_backup: Optional[str] = None


class CleanupPayload(BaseModel):
    older_than_days: int = Field(default=30, ge=1)
    dry_run: bool = True


class CleanupOut(BaseModel):
    dry_run: bool
    files_to_delete: int
    total_size_mb: float
    older_than_days: int
    message: str


class ResetDemoPayload(BaseModel):
    confirm: bool = False
    include_real_runs: bool = False
    include_review_data: bool = False


class ResetDemoOut(BaseModel):
    status: str
    message: str
    deleted_count: int
    pre_reset_backup: Optional[str] = None


class SnapshotOut(BaseModel):
    snapshot_path: str
    size_mb: float
    created_at: str
    manifest: dict
